exasol_advanced_analytics_framework/interface/create_event_handler_udf.py

```diff
@@ -22,58 +24,85 @@ def __init__(self, exa):
+            iter_num, event_handler_class,
+            bucketfs_location, latest_bucketfs_path)
         event_handler_context: EventHandlerContext = latest_state.context
         event_handler: EventHandlerBase = latest_state.event_handler
 
         # call the user code
         udf_context = self._create_udf_context_wrapper(ctx)
         result: EventHandlerResult = event_handler.handle_event(
-            udf_context, event_handler_context)  # TODO does user change state?
+            udf_context, event_handler_context)
 
-        # update and save (overwrite)the current state
+        # save the current state
+        iter_num += 1
+        current_bucketfs_path = PurePosixPath(
+            f"{event_handler_class}_{str(iter_num)}.pkl")
         current_state = EventHandlerState(event_handler_context, event_handler)
         self._save_current_state(
-            bucketfs_location, bucketfs_path, current_state)
+            bucketfs_location, current_bucketfs_path, current_state)
+
+        # remove previous state
+        self._remove_previous_state(
+            bucketfs_location, latest_bucketfs_path)
+
+        # wrap return query
+        return_query_view, return_query = self._wrap_return_query(
+            iter_num, bucketfs_connection,
+            result.return_query, result.return_query_columns)
 
         # return queries
-        ctx.emit(result.return_query)
+        ctx.emit(return_query_view)  # TODO-4
+        ctx.emit(return_query)
         ctx.emit(result.status)
         for query in result.query_list:
             ctx.emit(query)
 
+    @staticmethod
     def _load_latest_state(
-            self,
-            parameters: dict,
+            iter_num: int,
+            event_handler_class: str,
             bucketfs_location: BucketFSLocation,
             bucketfs_path: PurePosixPath) -> EventHandlerState:
 
-        event_handler_state = bucketfs_location.\
-            download_object_from_bucketfs_via_joblib(str(bucketfs_path))
-
-        if not event_handler_state:
+        if iter_num > 0:
+            # load the latest state
+            event_handler_state = bucketfs_location.\
+                download_object_from_bucketfs_via_joblib(str(bucketfs_path))
+        else:
+            # create the new state
             context = EventHandlerContext(
                 bucketfs_location, bucketfs_path)
             event_handler_class = getattr(importlib.import_module(
-                parameters["module_name"]), parameters["class_name"])
+                'module_name'), event_handler_class)  # TODO-1: module name how to get
             event_handler_obj = event_handler_class()
             event_handler_state = EventHandlerState(context, event_handler_obj)
 
         return event_handler_state
 
+    @staticmethod
+    def _remove_previous_state(
+            bucketfs_location: BucketFSLocation,
+            bucketfs_path: PurePosixPath) -> None:
+        # TODO-2: requests.delete(url, ..)
```

Comment:
bucketfs-utils does not provide any 'delete'  method. Should we send an explicit delete request to Bucketfs or is there any better way to do it?